src/SFA.DAS.Payments.MatchedLearner.Data/Repositories/MatchedLearnerRepository.cs

```diff
@@ -62,6 +64,8 @@ public async Task<MatchedLearnerDataLockInfo> GetDataLockEvents(long ukprn, long
 
             var transactionTypes = new List<byte> { 1, 2, 3 };
 
+            var latestProviderSubmittedJob = await GetLatestSubmissionJobForProvider(ukprn);
```

Comment:
not sure how it was handled in previous code but we should add a check where we can find a provide but we cant fine a learner in that case we should return 404